observer/lib/observer.js

```diff
@@ -37,3 +49,34 @@ export const observeTransferEvents = async (pgPool, ieContract, provider) => {
     await updateDailyTransferStats(pgPool, transferEvent, currentBlockNumber)
   }
 }
+
+/**
+ * Observe scheduled rewards on the Filecoin blockchain
+ * @param {import('pg').Pool} pgPool
+ * @param {import('ethers').Contract} ieContract
+ */
+const observeScheduledRewards = async (pgPool, ieContract) => {
+  console.log('Querying scheduled rewards from impact evaluator')
+  for (let i = 0; ; i++) {
+    let address
+    try {
+      address = await ieContract.readyForTransfer(i)
```

Comment:
IIRC, this will not find all participant addresses, only those with more than 0.1 FIL in scheduled rewards.

IMO, the ideal solution is to improve the impact evaluator smart contract to emit a new event whenever the amount of participant's scheduled rewards change. 

Since we don't have that event yet, can we perhaps query the blockchain history to find the `setScores` calls made by spark-evaluate for the last round? The second argument of that function contains a list of participant addresses.

https://github.com/Meridian-IE/impact-evaluator/blob/087c50d401ed7d67de0229bc227f0149764d0671/src/ImpactEvaluator.sol#L118-L122

Reply:
Fortunately, we already have `daily_participants` table that's populated by `spark-evaluate` and that we can access from spark-stats.

I propose to rework the part getting the list of participant addresses to query that table. For example:

```sql
SELECT participant_address
FROM daily_participants
LEFT JOIN participants
ON participant_id = id
WHERE day = now()
```

Reply:
I posted my comment before reading your two comments above.

I replied to your comments here:
https://github.com/filecoin-station/spark-stats/pull/131#issuecomment-2156444081

Sorry for the mess. 🙈